iodata/test/test_cube.py

```diff
@@ -66,6 +66,24 @@ def test_load_dump_load_aelta(tmpdir):
     dump_one(mol1, fn_cube2)
     mol2 = load_one(fn_cube2)
 
+    with open(fn_cube2) as f:
+        line_counter = 0
+        block_counter = 0
+        for line in f:
+            line_counter += 1
+            if line_counter > 6 + len(mol2.atnums):
+                if mol2.cube.shape[2] % 6 == 0:
+                    assert len(line.split()) == 6
+                if mol2.cube.shape[2] % 6 != 0:
+                    block_line_counter = (
+                            line_counter - (6 + len(mol2.atnums)
+                            + block_counter * (mol2.cube.shape[2] // 6 + 1))
```

Comment:
This is the piece of code to which the error messages apply.

```
79:29     iodata/test/test_cube.py  E126 continuation line over-indented for hanging indent
80:29     iodata/test/test_cube.py  E128 continuation line under-indented for visual indent
```

The code on these two lines is indented incorrectly. There are many possible ways to fix this. If you have cardboardlint installed, just run `cardboardlinter -F`. For details see https://iodata.readthedocs.io/en/latest/contributing.html

This should be a correct indentation, which is also fairly readable.

```python
                    block_line_counter = (
                        line_counter - (
                            6 + len(mol2.atnums) +
                            block_counter * (mol2.cube.shape[2] // 6 + 1)
                        )
                    )
```

The basic rule is to always indent by four spaces when breaking a line right after an opening bracket (`[`, `(`, ...). More indentation should be added whenever wrapping lines after new opening brackets. When wrapping not directly after a bracket, indentation should match character after the last opening bracket that was not closed on the previous line. (I'm not sure if that is even proper English. I hope it makes sense.) These indentation rules are rather complicated at first but they can improve readability a lot.

Reply:
Thank you a lot for your explanations. I think I better understand it now, it should be ok now.